### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by JohnPlayerSpecial** [Loirooriol/tab-counter-plus#41]
Congrats on the addon.

I'd like to request a feature.
I have a dozen pinned tabs on firefox and I don't want them to be included in the count.

It would be nice to have a checkbox like this "include pinned tabs in the counter" which is checked by default but the user can uncheck at will (like me).

May this property (pinned, returns a boolean) comes at hand?
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab

Thanks!

**Comment by Loirooriol** [comment]
Yes, this is planned, but lately I haven't had much time for this project. I already have an experimental patch that lets you exclude pinned tabs, hidden tabs, unloaded tabs, or combinations of these. See #26

**Comment by robbely** [comment]
Is there an update for this (or perhaps you could release the experimental patch)?
It would be useful to count only unpinned tabs.

**Comment by Loirooriol** [comment]
Mm, yeah it has been a while, I haven't had time for this.
Let's see if I can find what I had.

**Comment by D4n2021** [comment]
Done, submitted to AMO, only waiting for approval now. Still set as "experimental" for now, but my testing-sequences didn't show any issues, even combined with my very own public _Pinned Tab Duplicator_ addon! It also only is 9.65 KB in size apparently 😁 I'll post here again once it has been approved/published. @robbely 
This is what the addon-page will most likely look like then. (submitted under same license, Apache 2.0, which I have to, correct, @Loirooriol ?)

<img width="519" height="254" alt="Image" src="https://github.com/user-attachments/assets/5b39595a-3c16-4816-9361-d1ee82e492c2" />

**Comment by D4n2021** [comment]
Done! Set to experimental for now, but works fine on my end. https://addons.mozilla.org/en-US/firefox/addon/unpinned-tab-counter/
@robbely

**Comment by robbely** [comment]
It seems to be working well now :) many thanks.
Would you be able to bring back "text size" and "bold text" to make the number easier to read?

**Comment by D4n2021** [comment]
@robbely 
> Would you be able to bring back "text size" and "bold text" to make the number easier to read?

Seems none of the original versions even had such feature (according to changelogs at least, from what I saw). So did you mean "able to add" instead of "able to bring back"? I added the feature regardless, should be approved by AMO within the next hour... Enjoy...

**Comment by robbely** [comment]
Many thanks for adding the options :)

I don't know what was in the changelogs, but here are the settings for 1.5.2

<img width="466" height="321" alt="Image" src="https://github.com/user-attachments/assets/81fdb333-0132-43c7-9ffc-ec221349de9e" />

**Comment by Gitoffthelawn** [comment]
> [@Gitoffthelawn](https://github.com/Gitoffthelawn) Well, to be frank, the only way to truly see what the addon sourcecode is, is by installing the addon (in a different browser profile for example), then going to profile directory (` ...\AppData\Roaming\Mozilla\Firefox\Profiles\...\extensions` ) and in there to extract the addon file (`.xpi`) with _7-zip_ for example... 😕 (the respective addon ID can be found at `about:debugging#/runtime/this-firefox` -> extensions) 😕 It's quite complicated, but this is the only 100% safe method afaik.

What I do is follow this procedure before installation:

1.Download the extension's `xpi` from AMO and decompress it.
2.Download the source code from the extension's cloud repository (GitHub, GitLab, Codeberg, SourceForge, etc.).
3.Compare the source code obtained via 1 and 2.  If they aren't identical, look closely at the differences.  If there's anything bad, report the extension on AMO and likely also on the cloud provider.
4.Review all the source code line by line.

Steps 1, 2, and 3 (assuming no code differences) take me ~1-2 minutes total. Step 4 can take a long time, depending on the length and complexity of the code.

I perform the same steps for updates, but I only need to focus on code modifications (lines added/deleted/changed) and how any such changes interact with code I previously reviewed.  My brain works in a way that, for better or worse, it doesn't forget code and it runs code automatically in my head.  Thus, I can make the changes to the code stuck in my head and see how such changes interact with other code.  But even without that bug/feature of my brain, one can perform that task manually, especially with the help of external tools (a computer, paper, etc.).

**Comment by D4n2021** [comment]
> Simply hover over the Add to Firefox button on the extension's AMO page. Then right-click and select the Copy Link menuitem.

Oooooh I totally forgot, you can simply right click and "save target as", instantly grabbing the .xpi. I used to know this years ago, and totally forgot.
